### PR TITLE
Lmb 976 | Burgemeester selector iv in service REFACTOR

### PR DESCRIPTION
**Pull request by JonasVanHoof**
## Description

We want to move all the logic that is used for the IV page to be in the installatievergadering service. This will remove some arguments and methods what should make the code better understandable.

## How to test

Everything should work as before.

## Links to other PR's

- /

### CODE REVIEW

File: app/routes/verkiezingen/installatievergadering.js [hunk 1]
```diff
@@ -45,29 +45,23 @@ export default class PrepareInstallatievergaderingRoute extends Route {
...
+
     const verkiezingen = await this.getVerkiezingen(selectedPeriod);
     const kandidatenlijsten = await this.getKandidatenLijsten(selectedPeriod);
 
     const mandatarisForm = getFormFrom(this.store, MANDATARIS_EXTENDED_FORM);
 
     return RSVP.hash({
-      ivStatuses,
-      installatievergadering,
+      installatievergadering: this.ivService.iv,
```
Comment (JonasVanHoof) [inline app/routes/verkiezingen/installatievergadering.js:64]: remove this*

File: app/routes/verkiezingen/installatievergadering.js [hunk 2]
```diff
@@ -45,29 +45,23 @@ export default class PrepareInstallatievergaderingRoute extends Route {
...
-
     const bestuursorganenInTijd =
       await this.getBestuursorganenInTijd(selectedPeriod);
 
+    await this.ivService.setup(selectedPeriod);
+    if (this.ivService.iv) {
+      await this.ivService.createBestuursorganenInTijdMap(
+        bestuursorganenInTijd
+      );
+    }
```
Comment (JonasVanHoof) [inline app/routes/verkiezingen/installatievergadering.js:56]: Look where we use the isBehandeld and else just add the bestuursorganen in tijd to the setup method so we lose this if statement.

File: app/routes/verkiezingen/installatievergadering.js [hunk 3]
```diff
@@ -45,29 +45,23 @@ export default class PrepareInstallatievergaderingRoute extends Route {
...
     const kandidatenlijsten = await this.getKandidatenLijsten(selectedPeriod);
 
     const mandatarisForm = getFormFrom(this.store, MANDATARIS_EXTENDED_FORM);
 
     return RSVP.hash({
-      ivStatuses,
-      installatievergadering,
+      installatievergadering: this.ivService.iv,
       bestuurseenheid,
       bestuursorganenInTijd,
```
Comment (JonasVanHoof) [inline app/routes/verkiezingen/installatievergadering.js:66]: This can also be retrieved from the service*

**Review by JonasVanHoof — COMMENTED** [review]
TODO

File: app/routes/verkiezingen/installatievergadering.js (hunk 2 above)
Comment (JonasVanHoof) [reply app/routes/verkiezingen/installatievergadering.js]: removed it added the bois to the setup method

**Review by JonasVanHoof — COMMENTED** [review]
Personal comments

File: app/components/verkiezingen/draft-mandataris-list.hbs [hunk 4]
```diff
@@ -32,63 +32,86 @@
...
+                <td>
+                  {{mandataris.heeftLidmaatschap.binnenFractie.naam}}
+                </td>
+              {{/if}}
+              {{#if @showFunctie}}
+                <td>
+                  {{mandataris.bekleedt.bestuursfunctie.label}}
+                </td>
+              {{/if}}
+              {{#if @showRangorde}}
```
Comment (JonasVanHoof) [inline app/components/verkiezingen/draft-mandataris-list.hbs:74]: This didn't change only added the loaders at the top

File: app/components/verkiezingen/draft-mandataris-list.js [hunk 5]
```diff
@@ -33,21 +34,21 @@ export default class DraftMandatarisListComponent extends Component {
   }
 
   get resortedMandatarissen() {
-    return orderMandatarissenByRangorde(
-      [...this.mandatarissen],
-      this.installatievergadering.sortedMandatarissen
-    );
+    const m = [...this.args.mandatarissen];
+    return orderMandatarissenByRangorde(m, []);
```
Comment (JonasVanHoof) [inline app/components/verkiezingen/draft-mandataris-list.js:38]: still not sure if it made a difference?

**Comment by Rahien** [comment]
Closing this one. We learned a lot from this refactor but the gain doesn't outweigh the efforts/risks right now as the IV module will be disabled at the end of januari anyway

**Pull request closed without merging**